src/components/ConfigEditor/ConnectionConfig.tsx

```diff
@@ -139,7 +139,7 @@ const ConnectionConfig: React.FC<ConnectionConfigProps> = ({
         inputWidth={30}
         placeholder=""
         onReset={() => handleClearClientSecret()}
-        onChange={handleClientSecretChange}
+        onBlur={handleClientSecretChange}
         isConfigured={!!secureJsonData?.clientSecret || !!secureJsonFields?.clientSecret}
```

Comment:
The problem with this fix is that if the user clicks outside the input while they are still typing it will hit the bug anyway. What I would do is change the condition `isConfigured` to only be true if the `clientSecret` has some value when first loading the component.

(another reason to keep working on https://github.com/grafana/grafana/issues/33809 since this bug is not present for Azure Monitor)